arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -408,7 +390,7 @@ int arm_core_mpu_buffer_validate(void *addr, size_t size, int write)
 	u32_t r_index;
 
 	/* Iterate all mpu regions */
-	for (r_index = 0; r_index < _get_num_regions(); r_index++) {
+	for (r_index = 0; r_index < _get_num_usable_regions(); r_index++) {
```

Comment:
_get_num_regions() or _get_num_usable_regions() unfortunately can't be marked with __attribute__((pure)), so the compiler will most likely keep calling them.  Wouldn't be better to cache their value in an `u8_t`, and define `r_index` as an `u8_t` as well so that they're of the same type?

Reply:
done